geniza/corpus/templatetags/corpus_extras.py

```diff
@@ -117,3 +119,21 @@ def h1_to_h3(html):
     """Convert h1 headers to h3 to match other transcription formats,
     used to avoid modeltranslation inserting elements into h1 headers"""
     return html.replace("h1", "h3")
+
+
+def replacement_link(matchobj):
+    """Given a regex match \"PGPID #\", return a link to the referenced
+    document with the original string as link text."""
+    try:
+        doc = Document.objects.get(id=int(matchobj.group(1)))
+        doc_link = doc.permalink
+        return '<a href="%s">%s</a>' % (doc_link, matchobj.group(0))
+    except ObjectDoesNotExist:
+        return matchobj.group(0)
+
+
+@register.filter
+def pgp_urlize(text):
+    """Find all instances of \"PGPID #\" in the passed text, and convert
+    each to a link to the referenced document."""
```

Comment:
suggestion: generate a placeholder link using `reverse`, replace the id with a regex replacement string, and construct the link so that all matches can be replaced in a single regex.